File: Objc/PixelPixie/Node/PPStatNode.m

#import "PPStatNode.h"

@implementation PPStatNode

SKScene * parentScene; // used for jump scene
PPSpriteButton * btAgain, * bt2Menu;

- (instancetype)initWithScene:(SKScene *)scene Data:(PPData *)data {
    if (self = [super init]) {
        parentScene = scene;
        
        self.size = CGSizeMake(240, 360);
        self.color = [UIColor whiteColor];
        
        btAgain = [PPSpriteButton buttonWithColor:[UIColor redColor] andSize:CGSizeMake(100, 50)];
        btAgain.position = CGPointMake(50, -70);
        [btAgain setLabelWithText:@"AGAIN" withColor:[UIColor whiteColor]];
        [btAgain addTarget:self selector:@selector(clickAgain) withObject:nil forControlEvent:PPButtonControlEventTouchUpInside];
        [self addChild:btAgain];
        
        bt2Menu = [PPSpriteButton buttonWithColor:[UIColor blueColor] andSize:CGSizeMake(100, 50)];
        bt2Menu.position = CGPointMake(-50, -70);
        [bt2Menu setLabelWithText:@"MENU" withColor:[UIColor whiteColor]];
        [bt2Menu addTarget:self selector:@selector(click2Menu) withObject:nil forControlEvent:PPButtonControlEventTouchUpInside];
        [self addChild:bt2Menu];
    }
    return self;
}

- (void)click2Menu {
    [PPSceneManager jumpFrom:parentScene to:[PPSceneManager getStartScene]];
}

- (void)clickAgain {
    [PPSceneManager jumpFrom:parentScene to:[PPSceneManager getAnotherGameScene]];
    
}

@end

File: Objc/PixelPixie/Node/PPStatNode.swift
//
//  PPStatNode.swift
//  PixelPixie
//
//  Created by Junchao Yu on 2018/11/17.
//  Copyright © 2018年 silver6wings. All rights reserved.
//

@objc
class PPStatNode: SKSpriteNode {
    weak var parentScene: SKScene? = nil
    var btAgain: PPSpriteButton = PPSpriteButton(color: UIColor.red, size: CGSize(width: 100, height: 50))
    var bt2Menu: PPSpriteButton = PPSpriteButton(color: UIColor.blue, size: CGSize(width: 100, height: 50))
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    init(scene: SKScene, data: PPData) {
        super.init(texture: nil, color: UIColor.white, size: CGSize(width: 240, height: 360))
        
        self.parentScene = scene
        
        self.btAgain.position = CGPoint(x: 50, y: -70)
        self.btAgain.setLabelWithText("AGAIN", with: UIColor.white)
        self.btAgain.addTarget(self, selector: #selector(clickAgain), with: nil, for: PPButtonControlEvent.touchUpInside)
        self.addChild(self.btAgain)
        
        self.bt2Menu.position = CGPoint(x: -50, y: -70)
        self.bt2Menu.setLabelWithText("MENU", with: UIColor.white)
        self.bt2Menu.addTarget(self, selector: #selector(click2Menu), with: nil, for: PPButtonControlEvent.touchUpInside)
        self.addChild(self.bt2Menu)
    }
    
    func click2Menu() -> Void {
        PPSceneManager.jump(from: self.parentScene!, to: PPSceneManager.getStartScene())
    }
    
    func clickAgain() -> Void {
        PPSceneManager.jump(from: self.parentScene!, to: PPSceneManager.getAnotherGameScene())
    }
    
}
